companion/lib/Service/RegexRouter.ts

```diff
@@ -51,25 +49,10 @@ export class RegexRouter {
 	/**
 	 * Add a route to the router
 	 */
-	addRegex(regexp: RegExp, handler: RouteHandler): void {
-		if (!regexp || !handler) throw new Error('Invalid route parameters')
-		this.#routes.push({ regexp, handler })
-	}
-
-	/**
-	 * Add a route to the router, using the express style path syntax
-	 */
-	addPath(path: string, handler: PathRouteHandler): void {
```

Comment:
I'm wondering whether we should be keeping pathToRegexp and this method to use for the osc api. Because that api is using typical slash separated components, so wouldn't be fighting the library?

I don't know though, what do you think?

Reply:
My taste -- I emphasize this is a _taste_ as this is your turf :slightly_smiling_face: -- is to prefer built-in language function over libraries, when they do similar things.  Regular expression syntax is in the JS author's toolkit.  A new regular expression language is not.  (And theoretically could have its own corner cases.)

I also prefer there be fewer ways to do it, not more.  Two distinct pattern mechanisms seems worse than one -- perhaps than either one.

Regular expression pathological cases happen when you have repetition with possibility of matching multiple ways.  None of these cases have that.  ReDOS simply isn't a worry here.

Honestly, tho, sequentially testing a list of regular expressions isn't the right way for either of these cases compared to actual parsing.  For OSC I'd `address.split('/')` and iterate through the components.  For TCP I'd incrementally tokenize the string.  I'd use some `switch`es to invoke the intended operation -- and return informative errors for contextual errors.

But I'd never start writing that patch if I didn't know it'd be accepted.  :slightly_smiling_face:  Regular expressions aren't right for this, but for a fixed list of a couple dozen syntaxes they're bearable, in a patch with the least code most likely to be accepted.